resources/js/confirm-delete.js

```diff
@@ -1,10 +1,45 @@
+// Show confirm dialog before submitting any spoofed DELETE method form
 document.body.addEventListener("submit", function(event) {
-  if (event.target.querySelector('input[name="_method"][value="delete"], input[name="_method"][value="DELETE"]')) {
+  let method;
+  let deleteButton;
+
+  // Take last spoofed method using FormData if available
+  if (FormData instanceof Object) {
+    let formData = new FormData(event.target);
+    if (typeof formData.getAll == "function") {
+      method = (
+        formData.getAll("_method").pop() || event.target.method
```

Comment:
@bjuppa Two questions here:

1. Will there always just be one spoofed method per submit? 
2. I suspect the event.target.method parameter will still be set whether or not the FormData object is available or not? If so we could perhaps do that check after the if-statement so everything in there is just related to the FormData object, and set the method-variable to that value after if it wasn't set?



1.

Reply:
1. Actually there could be more than one spoofed method even though it'd be uncommon! I can think of a case where the main method is PATCH and there's also a spoofed method DELETE button. If more than one input with same name is present in the form, the last one submitted is the value received by the server, so that's why I chose to pop the last one off the end of the array. And the array in this case should be what is actually posted, so a not-clicked button should not be in the array.
2. I wanted to leave the method variable empty if the browser doesn't support `FormData`, to be able to extract it using form `elements` later. So, if browser supports `FormData` the "real" method is either in the last `_method` input of the "submitted data" or else it's the forms method (usually POST, and that makes the rest of this script do nothing). If the browser doesn't support `FormData` the method should be extracted from the first `_method` input, so that it doesn't extract it from any submit button or other not-activated input that has another `_method` than the forms main method. This could make the delete alert not pop up for some cases in an old browser where there's a `_method="DELETE"` button in the form. But it would be more annoying if the delete warning pope up on non-delete actions in modern browsers 😄 
So long story short, by not setting the method variable at all, we indicate to the later parts of the script that we haven't been able to extract the spoofed method safely.